File: app/src/main/java/com/cylonid/nativealpha/AboutActivity.java
package com.cylonid.nativealpha;


import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import com.mikepenz.aboutlibraries.LibsBuilder;

import java.time.Year;

import mehdi.sakout.aboutpage.AboutPage;
import mehdi.sakout.aboutpage.Element;

public class AboutActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        View aboutPage = new AboutPage(this)
                .enableDarkMode(false)
                .setDescription("Native Alpha for Android\nby cylonid © " + Year.now().getValue())
                .setImage(R.drawable.native_alpha_foreground)
                .addItem(new Element().setTitle("Version " + BuildConfig.VERSION_NAME))
                .addGitHub("cylonid", "Find us on GitHub")
                .addPlayStore("com.cylonid.nativealpha")
                .addGroup(getString(R.string.eula_title))
                .addItem(showEULA())
                .addGroup(getString(R.string.license))
                .addItem(showLicense())
                .addItem(showOpenSourcelibs())
                .create();

        setContentView(aboutPage);
    }

    Element showEULA() {
        Element license = new Element();
        license.setTitle(getString(R.string.eula_content));
        return license;
    }
    Element showLicense() {
        Element license = new Element();

        license.setTitle(getString(R.string.gnu_license));
        license.setOnClickListener(v -> {
            String url = "https://www.gnu.org/licenses/gpl-3.0.txt";
            Intent i = new Intent(Intent.ACTION_VIEW);
            i.setData(Uri.parse(url));
            startActivity(i);
        });
        return license;

    }
    Element showPayPal() {
        Element license = new Element();

        license.setTitle(getString(R.string.paypal));
        license.setOnClickListener(v -> {
            String url = "https://paypal.me/cylonid";
            Intent i = new Intent(Intent.ACTION_VIEW);
            i.setData(Uri.parse(url));
            startActivity(i);
        });
        return license;

    }

    Element showOpenSourcelibs() {
        Element os = new Element();
        os.setTitle(getString(R.string.open_source_libs));
        os.setOnClickListener(v -> {
            startActivity(new LibsBuilder()
                    .withEdgeToEdge(true)
                    .withSearchEnabled(true)
                    .intent(this));
        });
        return os;
    }



//    Element getCopyRightsElement() {
//        Element copyRightsElement = new Element();
//        final String copyrights = String.format(getString(R.string.copy_right), Calendar.getInstance().get(Calendar.YEAR));
//        copyRightsElement.setTitle(copyrights);
//        copyRightsElement.setIconDrawable(R.drawable.about_icon_copy_right);
//        copyRightsElement.setAutoApplyIconTint(true);
//        copyRightsElement.setIconTint(mehdi.sakout.aboutpage.R.color.about_item_icon_color);
//        copyRightsElement.setIconNightTint(android.R.color.white);
//        copyRightsElement.setGravity(Gravity.CENTER);
//        copyRightsElement.setOnClickListener(new View.OnClickListener() {
//            @Override
//            public void onClick(View v) {
//                Toast.makeText(AboutActivity.this, copyrights, Toast.LENGTH_SHORT).show();
//            }
//        });
//        return copyRightsElement;
//    }
}

File: app/src/main/java/com/cylonid/nativealpha/AboutActivity.kt
package com.cylonid.nativealpha

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.View
import androidx.activity.addCallback
import androidx.appcompat.app.AppCompatActivity
import com.cylonid.nativealpha.databinding.ActivityToolbarBaseBinding
import com.cylonid.nativealpha.util.ColorUtils.getColorResFromThemeAttr
import com.mikepenz.aboutlibraries.LibsBuilder
import mehdi.sakout.aboutpage.AboutPage
import mehdi.sakout.aboutpage.Element
import java.time.Year
import com.cylonid.nativealpha.util.WindowInsetsUtils

class AboutActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val baseBinding = ActivityToolbarBaseBinding.inflate(layoutInflater)
        setContentView(baseBinding.root)
        WindowInsetsUtils.applyAsPadding(this, false)

        baseBinding.activityContent.addView(generateAboutPageView())

        val toolbar = baseBinding.toolbar.topAppBar
        setSupportActionBar(toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.title = getString(R.string.app_info)

        onBackPressedDispatcher.addCallback(this) {
            finish()
        }

        toolbar.setNavigationOnClickListener {
            onBackPressedDispatcher.onBackPressed()
        }

    }

    private fun generateAboutPageView(): View {
        val page = AboutPage(this).apply {
            setDescription(
                """
                Native Alpha for Android
                by cylonid © ${Year.now().value}
                """.trimIndent()
            )
            setImage(R.drawable.native_alpha_foreground)
            addItem(Element().setTitle("Version " + BuildConfig.VERSION_NAME))
            addItem(addGitHubCustom("cylonid", "GitHub"))
            addPlayStore("com.cylonid.nativealpha.pro", "Play Store")
            addWebsite(
                "https://github.com/cylonid/NativeAlphaForAndroid/blob/dev/privacy_policy.md",
                getString(
                    R.string.privacy_policy
                )
            )
            if(BuildConfig.FLAVOR == "extendedGithub") {
                addItem(showLiberaPay())
            }
            addGroup(getString(R.string.eula_title))
            addItem(showEULA())
            addGroup(getString(R.string.license))
            addItem(showLicense())
            addItem(showOpenSourcelibs())
        }

        return page.create()

    }

    private fun addGitHubCustom(id: String, title: String): Element {
        val gitHubElement = Element()
        gitHubElement.setTitle(title)
        gitHubElement.setIconDrawable(R.drawable.about_icon_github)
        gitHubElement.setIconTint(
            getColorResFromThemeAttr(
                this, com.google.android.material.R.attr.colorOnSurface, R.color.about_github_color
            )
        )
        gitHubElement.setIconNightTint(R.color.about_item_dark_text_color)
        gitHubElement.setValue(id)

        val intent = Intent()
        intent.setAction(Intent.ACTION_VIEW)
        intent.addCategory(Intent.CATEGORY_BROWSABLE)
        intent.setData(Uri.parse(String.format("https://github.com/%s", id)))

        gitHubElement.setIntent(intent)

        return gitHubElement
    }

    fun showEULA(): Element {
        val license = Element()
        license.setTitle(getString(R.string.eula_content))
        return license
    }

    fun showLicense(): Element {
        val license = Element()

        license.setTitle(getString(R.string.gnu_license))
        license.setOnClickListener {
            val url = "https://www.gnu.org/licenses/gpl-3.0.txt"
            val i = Intent(Intent.ACTION_VIEW)
            i.setData(Uri.parse(url))
            startActivity(i)
        }
        return license
    }

    fun showLiberaPay(): Element {
        val element = Element()
        element.setTitle(getString(R.string.support_liberapay))
        element.setIconDrawable(R.drawable.liberapay_logo)
        element.skipTint = true

        element.setOnClickListener {
            val url = "https://liberapay.com/cylonid"
            val i = Intent(Intent.ACTION_VIEW)
            i.setData(Uri.parse(url))
            startActivity(i)
        }
        return element
    }

    fun showPayPal(): Element {
        val license = Element()

        license.setTitle(getString(R.string.paypal))
        license.setOnClickListener {
            val url = "https://paypal.me/cylonid"
            val i = Intent(Intent.ACTION_VIEW)
            i.setData(Uri.parse(url))
            startActivity(i)
        }
        return license
    }

    fun showOpenSourcelibs(): Element {
        val os = Element()
        os.setTitle(getString(R.string.open_source_libs))
        os.setOnClickListener {
            startActivity(
                LibsBuilder()
                    .withEdgeToEdge(true)
                    .withSearchEnabled(true)
                    .intent(this)
            )
        }
        return os
    }
}